### PR TITLE
Added AutoGPTQ readme and test

### PR DESCRIPTION
**Pull request by HolyFalafel**
[Added AutoGPTQ UINT4 to README.md (](https://github.com/huggingface/optimum-habana/commit/1e3ccc0fa5afae12cceea45868d8647ef9e48d33)https://github.com/huggingface/optimum-habana/pull/279[)](https://github.com/huggingface/optimum-habana/commit/1e3ccc0fa5afae12cceea45868d8647ef9e48d33)

[Added SRAM_SLICER_SHARED_MME_INPUT_EXPANSION_ENABLED envar to the readme, in AutoGPTQ](https://github.com/huggingface/optimum-habana/commit/6790dd2cff481c31d810ae191334438512089ced) 

(https://github.com/huggingface/optimum-habana/pull/305)

[Added gptq to test_text_gen (](https://github.com/huggingface/optimum-habana/commit/a9f4b5496cbfeb3f1c51ce2b0402775eb44cb62f)https://github.com/huggingface/optimum-habana/pull/307[)](https://github.com/huggingface/optimum-habana/commit/a9f4b5496cbfeb3f1c51ce2b0402775eb44cb62f)

### CODE REVIEW

File: examples/text-generation/README.md [hunk 1]
```diff
@@ -494,6 +494,37 @@ python ../gaudi_spawn.py --use_deepspeed --world_size 8 run_generation.py \
 
 For more details see [documentation](https://docs.habana.ai/en/latest/PyTorch/Model_Optimization_PyTorch/Optimization_in_PyTorch_Models.html#using-fused-sdpa).
 
+### Running with UINT4 weight quantization using AutoGPTQ
+
+
+Llama2-7b in UINT4 weight only quantization is enabled using [AutoGPTQ Fork](https://github.com/HabanaAI/AutoGPTQ), which provides quantization capabilities in PyTorch.
+Currently, the support is for UINT4 inference of pre-quantized models only.
+
```
Comment (libinta) [inline examples/text-generation/README.md:502]: what does the pre-quantized model mean? By which process to pre-quantized it and with which precision?

Comment (HolyFalafel) [reply examples/text-generation/README.md:502]: It means that we currently don't support the quantization process. but only support loading an existing quantized model

**Review by MrGeva — COMMENTED** [review]
DO NOT merge it yet, need to get approval first

**Comment by vidyasiv** [comment]
@HolyFalafel , is this PR ready for v1.17? Please resolve conflicts against latest main.

**Comment by vidyasiv** [comment]
> DO NOT merge it yet, need to get approval first

@MrGeva Can you clarify if this is targeted for v1.17?

**Comment by emascarenhas** [comment]
Please sync your PR with main/upstream and fix any merge conflicts. Thank you.

**Comment by yuanwu2017** [comment]
@HolyFalafel @libinta 
Will this patch be merged? At present, Gaudi has been integrated into AutoGPTQ. But I didn't see OH to install this package in this patch. I made a [patch ](https://github.com/huggingface/tgi-gaudi/pull/217)to integrate the AutoGPTQ in tgi-gaudi and submitted a [patch ](https://github.com/huggingface/optimum/pull/2003) for optimum. It works. Can you add  AutoGPTQ package installation in this patch?

**Comment by emascarenhas** [comment]
Can you rebase and revise your patch if necessary, so that it merges cleanly with main?

Also do "pip install -U ruff; make style" and check for any issues.

In addition, please run "tests/ci/fast_tests.sh" after installing 
pip install -e ".[tests]" 
and any slow tests, e.g., 
GAUDI2_CI=1 RUN_SLOW=1 python -m pytest
 tests/test_text_generation_example.py -v -s -k
and report the results here.

**Comment by imangohari1** [comment]
> Can you rebase and revise your patch if necessary, so that it merges cleanly with main?
> 
> Also do "pip install -U ruff; make style" and check for any issues.
> 
> In addition, please run "tests/ci/fast_tests.sh" after installing pip install -e ".[tests]" and any slow tests, e.g., GAUDI2_CI=1 RUN_SLOW=1 python -m pytest tests/test_text_generation_example.py -v -s -k and report the results here.

@HolyFalafel 
Thanks for this draft PR. 
what is the status of this, if this is still needed:  Can you rebase, synch with top of main habana and follow the instructions to test it?
if not needed, please close. 
thanks.

**Pull request closed without merging**